### ISSUE TITLE
Is it a rank loss in source code?

### ISSUE THREAD

**Issue by DaoD** [pl8787/MatchPyramid-TensorFlow#3]
In the paper, u use a softmax loss for classification, but in the source code, it is the rank loss?
I'm not sure if I get the point.
Thanks

**Comment by pl8787** [comment]
Yes, that's right. It depends on what kind of task you have.

For classification task, use softmax loss.

For ranking task , use pair hinge loss(rank loss).

**Comment by DaoD** [comment]
@pl8787 Thx.
Another small question, what if a word doesn't have a trained embedding?  How to initialize it?
You know it is kind of difficult to have a embedding for every word by word2vec.

**Comment by pl8787** [comment]
@DaoD Just use Uniform Random [-0.2, 0.2].

**Comment by DaoD** [comment]
@pl8787 wow, thank you.
and could you plz explain the meaning of data_psize? Is it depends on the data max len? or anything else?

**Comment by pl8787** [comment]
@DaoD data_psize is the size of the pooled feature map.
